aws/resource_aws_s3control_bucket_lifecycle_configuration.go

```diff
@@ -0,0 +1,529 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/s3control"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsS3ControlBucketLifecycleConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsS3ControlBucketLifecycleConfigurationCreate,
+		Read:   resourceAwsS3ControlBucketLifecycleConfigurationRead,
+		Update: resourceAwsS3ControlBucketLifecycleConfigurationUpdate,
+		Delete: resourceAwsS3ControlBucketLifecycleConfigurationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"bucket": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+			"rule": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"abort_incomplete_multipart_upload": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"days_after_initiation": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+						"expiration": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"date": {
+										Type:     schema.TypeString,
+										Optional: true,
+										ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+											value := v.(string)
+
+											_, err := time.Parse("2006-01-02", value)
+
+											if err != nil {
+												errors = append(errors, fmt.Errorf("%q should be in YYYY-MM-DD date format", value))
+											}
+
+											return
+										},
+									},
```

Comment:
could also live in `validators.go` ? or possibly getting too large to keep populating ?

Reply:
Partially related to https://github.com/terraform-providers/terraform-provider-aws/issues/8424 -- although in this case its something that would ultimately be a good candidate for there (or a similar shared location) if it had more usage than just here and the `aws_s3_bucket` resource. 👍